File: Python/Peewee/models/employee.py
from enum import Enum
from uuid import uuid4

from models.base_model import BaseModel
from models.department import Department
from models.person import Person

from peewee import CharField, Field, ForeignKeyField


class Position(Enum):
    STAFFER = 'Staffer'
    MANAGER = 'Manager'
    BOSS = 'Boss'

    def __str__(self):
        return self.value


class PositionField(Field):
    field_type = CharField.field_type

    def db_value(self, value: Position):
        return value.value

    def python_value(self, value: str):
        return Position(value)


class Employee(BaseModel):
    id: str = CharField(primary_key=True, default=uuid4)
    position: Position = PositionField()
    person: Person = ForeignKeyField(Person, on_delete='CASCADE')
    department: Department = ForeignKeyField(Department, on_delete='CASCADE')

    class Meta:
        table_name = 'employees'
        model_name = 'Employee'

    def __str__(self):
        return (f'{{ Employee: {{ Id: {self.id}, Person: {self.person}, '
                f'Department: {self.department}, '
                f'Position: {self.position} }} }}')

File: TypeScript/MicroORM/models/Employee.ts
import { Entity, Enum, ManyToOne, PrimaryKey } from '@mikro-orm/core';
import { v4 } from 'uuid';
import { Department } from './Department';
import { Person } from './Person';

export enum Position {
    STAFFER = 'Staffer',
    MANAGER = 'Manager',
    BOSS = 'Boss',
}

@Entity({ tableName: 'employees' })
export class Employee {
    @PrimaryKey()
    public id = v4();

    @Enum({ items: () => Position, type: 'string' })
    public position!: Position;

    @ManyToOne({ onDelete: 'cascade' })
    public department!: Department;

    @ManyToOne({ onDelete: 'cascade' })
    public person!: Person;

    string(): string {
        return `{ Employee { Id: ${this.id}, Department: ${this.department.string()}, Person: ${this.person.string()}, Position: ${this.position} } }`;
    }
}
